### PR TITLE
[Feature] New privilege: authorization support for database object

### PR DESCRIPTION
**Pull request by nshangyiming**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Support authorization for database object, `create database xxx` will be added later when catalog object is ready.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] This pr will affect users' behaviors
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function: pm will add it all when 3.0 is released

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 79 / 86 (91.86%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterDatabaseRenameStatementAnalyzer.java|3|5|60.00%|[12, 16]|
|:large_blue_circle:|com/starrocks/sql/analyzer/DropStmtAnalyzer.java|3|4|75.00%|[88]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterDbQuotaAnalyzer.java|3|4|75.00%|[14]|
|:large_blue_circle:|com/starrocks/sql/analyzer/BasicDbStmtAnalyzer.java|15|17|88.24%|[17, 49]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|22|23|95.65%|[62]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterDatabaseQuotaStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/RecoverDbStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/DropDbStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeType.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/UseDbStmt.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterDatabaseRenameStatement.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowCreateDbStmt.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 79 / 88 (89.77%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterDatabaseRenameStatementAnalyzer.java|3|5|60.00%|[12, 16]|
|:large_blue_circle:|com/starrocks/sql/analyzer/DropStmtAnalyzer.java|3|4|75.00%|[88]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterDbQuotaAnalyzer.java|3|4|75.00%|[14]|
|:large_blue_circle:|com/starrocks/sql/analyzer/BasicDbStmtAnalyzer.java|15|17|88.24%|[17, 49]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|22|25|88.00%|[51, 63, 73]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterDatabaseQuotaStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/RecoverDbStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/DropDbStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeType.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/UseDbStmt.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterDatabaseRenameStatement.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowCreateDbStmt.java|3|3|100.00%|[]|

**Comment by nshangyiming** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java [hunk 1]
```diff
@@ -160,6 +179,47 @@ public Void visitTable(TableRelation node, Void context) {
...
+
+        @Override
+        public Void visitUseDbStatement(UseDbStmt statement, ConnectContext context) {
+            checkAnyActionInDb(context, statement.getCatalogName(), statement.getDbName());
+            return null;
+        }
+
+        @Override
+        public Void visitShowCreateDbStatement(ShowCreateDbStmt statement, ConnectContext context) {
+            checkAnyActionInDb(context, statement.getCatalogName(), statement.getDb());
```
[the remark this answers is not in the document]
Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:203]: resovled

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 94 / 107 (87.85%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterDatabaseRenameStatementAnalyzer.java|3|5|60.00%|[12, 16]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeManager.java|7|10|70.00%|[628, 629, 630]|
|:large_blue_circle:|com/starrocks/sql/analyzer/DropStmtAnalyzer.java|3|4|75.00%|[88]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterDbQuotaAnalyzer.java|3|4|75.00%|[14]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|27|31|87.10%|[51, 63, 73, 84]|
|:large_blue_circle:|com/starrocks/sql/analyzer/BasicDbStmtAnalyzer.java|15|17|88.24%|[17, 49]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterDatabaseQuotaStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/RecoverDbStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/DropDbStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeType.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/UseDbStmt.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterDatabaseRenameStatement.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/authentication/AuthenticationManager.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowCreateDbStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|2|2|100.00%|[]|

**Comment by nshangyiming** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 131 / 166 (78.92%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|1|00.00%|[2975]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterDatabaseRenameStatementAnalyzer.java|3|5|60.00%|[12, 16]|
|:large_blue_circle:|com/starrocks/mysql/privilege/AuthUpgrader.java|38|62|61.29%|[122, 154, 178, 220, 224, 265, 269, 296, 336, 373, 414, 424, 457, 487, 491, 540, 550, 596, 606, 677, 693, 707, 747, 766]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeManager.java|7|10|70.00%|[632, 633, 634]|
|:large_blue_circle:|com/starrocks/sql/analyzer/DropStmtAnalyzer.java|3|4|75.00%|[88]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterDbQuotaAnalyzer.java|3|4|75.00%|[14]|
|:large_blue_circle:|com/starrocks/sql/analyzer/BasicDbStmtAnalyzer.java|15|17|88.24%|[17, 49]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|26|27|96.30%|[105]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterDatabaseQuotaStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/RecoverDbStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/DropDbStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeType.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/UseDbStmt.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterDatabaseRenameStatement.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/authentication/AuthenticationManager.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowCreateDbStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|2|2|100.00%|[]|

**Review by padmejin — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by chaoyli — APPROVED** [review]

**Pull request merged**